Add optional bootloader support

endian-albin opened the pull request:
- west.yml: add mcuboot
- pinetime and p8 dts: new partition layout for internal flash and external SPI flash memory
- add dfu.conf
- cmake: build with dfu.conf if environment variable DFU="on"
- cmake: build with logging.conf only if environment variable LOGGING="on"
- main: confirm new image automatically
- backlight: use all three LEDs instead of just one
- README: update documentation

If DFU is not set to "on", then all the code changes here should be ignored except:
- the backlight fix (thanks @jf002!)
- logging, which will now be disabled by default

sdorre reviewed (commented):
Missing the `dfu.conf` file ?

endian-albin commented:
> Missing the `dfu.conf` file ?

thanks, fixed.

endian-albin commented:
I've updated the partition layout and fixed the backlight regression. We still need `CONFIG_ARM_MPU=n` for the application not to crash when confirming the new image.

sdorre commented on app/hypnos/CMakeLists.txt:
```diff
@@ -7,8 +7,10 @@ list(APPEND BOARD_ROOT ${CMAKE_CURRENT_LIST_DIR}/..)
 # Config files
-IF("$ENV{LOGGING}" STREQUAL "off")
-  message(STATUS "LOGGING will be disabled.")
+IF("$ENV{DFU}" STREQUAL "on")
```
What about doing it independently ?
```
set(CONF_FILE ${CMAKE_CURRENT_LIST_DIR}/prj.c)
IF("$ENV{DFU}" STREQUAL "on")
  set(CONF_FILE ${CONF_FILE} ${CMAKE_CURRENT_LIST_DIR}/dfu.conf)
ENDIF()
IF("$ENV{LOGGING}" STREQUAL "on")
  set(CONF_FILE ${CONF_FILE} ${CMAKE_CURRENT_LIST_DIR}/logging.conf)
ENDIF()
```

endian-albin replied:
Any suggestion?

endian-albin replied:
It would be nice to be able to set any combination of envs/configs if we get more in the future.

sdorre replied:
try it out. 
It sets the main confi file anyways and depending on the ENV variables it adds up the necessary files.

endian-albin replied:
OK! It will disable logging by default but that's OK I guess.

endian-albin replied:
@sdorre, I've made changes again and force-pushed.  Feel free to press merge if it looks alright to you now.

sdorre commented on app/hypnos/src/backlight.c:
```diff
@@ -23,14 +25,18 @@ static bool backlight_enabled = false;
 {
 	backlight_dev = device_get_binding(BACKLIGHT_PORT);
-	gpio_pin_configure(backlight_dev, BACKLIGHT, GPIO_OUTPUT);
+	gpio_pin_configure(backlight_dev, BACKLIGHT_1, GPIO_OUTPUT);
```
So issue was floating state for the other backlight gpios ?

endian-albin replied:
I suppose so. In any case, the initial state of the leds were different after booting with or without mcuboot.
